readme.md

```diff
@@ -4,24 +4,26 @@
 
 A minimal and opinionated OpenAI client powered by fetch.
 
-Unfortunately, the official [openai-node](https://github.com/openai/openai-node) uses Axios, which only supports Node and is bloated.
+Unfortunately, the official [openai-node](https://github.com/openai/openai-node) package patches fetch in problematic ways and is quite bloated.
 
 ### Reasons to consider using `openai-fetch`:
 
+- You want a fast and small client that doesn't patch fetch
```

Comment:
Thanks, but this adds a ton of weight/dependencies and a shim to accomplish something that is just a simple fetch request in the case of this library. What would be the benefit of doing this?

Reply:
The weight exists when you download the package, but should not at runtime in this case. 

The "shim" name is a bit misleading – here it just configures the package (including your TypeScript types, which is the trickiest part) to use the builtin, web fetch types.

We use `node-fetch` (soon [undici](https://github.com/openai/openai-node/pull/402), which is the same as Node's builtin `fetch`) by default because `fetch` is still experimental in Node.

If you were to use a bundler with the code above (eg, with NextJS), you should see a fairly small addition to your bundle size due to this library (I'd be curious to see, haven't tried yet). 

Most of the code downloaded from npm is sourcemaps, source files, and .cjs + .mjs versions of the same thing (but the user will only use one).

There's also some code related to streaming and file uploads, of course. If you don't need that, you could shave a few kb by removing it.

Reply:
This package is meant to be "A minimal and opinionated OpenAI client powered by fetch." The whole point is to use native fetch without any polyfills/patches and be minimal. I look forward to when the official library checks these boxes and I can archive this repo :)

Reply:
Sure, to each their own.

Just to clarify, using `import 'openai/shims/web'` [**does not**](https://github.com/openai/openai-node/blob/master/src/_shims/web-runtime.ts) do any real polyfilling or patches. 

It merely disables importing node-fetch, sets up appropriate TypeScript types, and alerts the user if they're missing a required global.

> I look forward to when the official library checks these boxes 

To set expectations, we don't expect to rely on built-in `fetch` until at least mid-2026, since `openai` supports all Node LTS versions until EOL and will not rely on experimental flags (and `fetch` is still experimental in Node 20, which hits EOL in mid-2026).

Reply:
Based on my understanding, you could perhaps consider redoing this repo as essentially
```ts
import 'openai/shims/web'
export * from 'openai'
```
alongside a `postinstall` step to remove the `node-fetch` and related dependencies. 

I'm not sure what your bundle size requirements are, though, and whether that approach would suit them, given the streaming support etc.

In any case, I can see that our docs here have not been great (sorry about that!), so I think I'll add a section to the README clarifying how and why users can dodge node-fetch et al. You certainly haven't been the first person to find it annoying that we prioritize compatibility in this way.

In any case, thanks very much for the feedback!

Reply:
Appreciate the input, Alex && all your worn on making the official `openai` package awesome 😄 

Going to resolve this convo for now since it's not really the best place to discuss. Thanks!